From 1319cbffbddd30ebccf482e5efdbbbee75ca6fdc Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Tue, 16 Sep 2014 17:06:44 -0400
Subject: [usb] uas: uas_alloc_data_urb - Remove unnecessary use_streams check

Message-id: <1410887463-20674-333-git-send-email-dzickus@redhat.com>
Patchwork-id: 95321
O-Subject: [RHEL7 PATCH 332/591] uas: uas_alloc_data_urb: Remove unnecessary use_streams check
Bugzilla: 1110939
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1110939

commit c6d4579d4ba24c494d03daf656cd2ff2a9e683c6
Author: Hans de Goede <hdegoede@redhat.com>
Date:   Tue Nov 12 10:53:57 2013 +0100

    uas: uas_alloc_data_urb: Remove unnecessary use_streams check

    uas_alloc_data_urb always gets called with a stream_id value of 0 when not
    using streams. Removing the check makes it consistent with uas_alloc_sense_urb.

    Signed-off-by: Hans de Goede <hdegoede@redhat.com>
    Signed-off-by: Sarah Sharp <sarah.a.sharp@linux.intel.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/usb/storage/uas.c b/drivers/usb/storage/uas.c
index 10e580e..e06505c 100644
--- a/drivers/usb/storage/uas.c
+++ b/drivers/usb/storage/uas.c
@@ -429,8 +429,7 @@ static struct urb *uas_alloc_data_urb(struct uas_dev_info *devinfo, gfp_t gfp,
   goto out;
  usb_fill_bulk_urb(urb, udev, pipe, NULL, sdb->length,
      uas_data_cmplt, cmnd);
- if (devinfo->use_streams)
-  urb->stream_id = stream_id;
+ urb->stream_id = stream_id;
  urb->num_sgs = udev->bus->sg_tablesize ? sdb->table.nents : 0;
  urb->sg = sdb->table.sgl;
  out:
-- 
1.7.1